python/services/evaluationservice/dmod/evaluationservice/evaluation_service/views/launch.py

```diff
@@ -51,30 +65,50 @@ def post(self, request, *args, **kwargs):
             "instructions": instructions
         }
         connection = utilities.get_redis_connection()
-        connection.publish("evaluation_jobs", json.dumps(launch_parameters))
+        connection.xadd(application_values.EVALUATION_QUEUE_NAME, launch_parameters)
 
         return response
 
 
+def get_evaluation_template() -> str:
+    """
+    Get raw text to serve as a starting point for an evaluation
+    """
+    with open(EVALUATION_TEMPLATE_PATH, "r") as evaluation_template_file:
+        return evaluation_template_file.read()
+
+
+def generate_evaluation_id() -> str:
```

Comment:
First question and it might be moot depending on the answer. Could / will a generated evaluation id ever make it into the db? I have a recollection that this was just for front end purposes, but just wanted to double check. If it is only for the front end, I think its probably fine. I would rather just return `current_date.isoformat(timespec="seconds")`, but honestly that is more of a nit than anything. If it could go into the db at some point:

My concern with this is that it is not unique enough. What are your thoughts on doing something like:

```python
import secrets
from datetime import datetime

def generate_evaluation_id() -> str:
  now = datetime.now()
  salt = "%05x" % secrets.randombits(20)
  time = now.isoformat(timespec="seconds")
  # ex: '2024-08-06T10:49:04_f8c61'
  return f"{time}_{salt}"
```

The small amount of salt will decrease the chance of collisions. Its not perfect and using something like a `ulid` would be preferred, but just wanted to put it out there.

Reply:
It's just a human friendly default/example identifier, not anything for uniqueness. It'll show up on the screen in the input box and if a user is going to be lazy, it'll submit the job and name it by that name. _Ideally_, the user would change the value to be more reflective of their work.

It is there so if you looked at a list of all the evaluations run in the last x hours, you'd be able to find the one you named. Hashes, random bits, etc don't work great in this context because the average goober doesn't naturally link that sort of data to concrete entities, though natural language and dates usually do the trick.

A docker container name-like scheme might work better, but that would require the involvement of random phrase generators.

If it _were_ to go into a database, it'd be in some `name` field, not a key.